numpy/lib/histograms.py

```diff
@@ -1047,7 +1047,13 @@ def histogramdd(sample, bins=10, range=None, normed=None, weights=None,
                 raise ValueError(
                     '`bins[{}]` must be positive, when an integer'.format(i))
             smin, smax = _get_outer_edges(sample[:,i], range[i])
-            edges[i] = np.linspace(smin, smax, bins[i] + 1)
+            try:
+                n = operator.index(bins[i])
+            
+            except TypeError as e:
+                raise TypeError("bins[i] must be an integer") from e
```

Comment:
```suggestion
                raise TypeError("when a scalar, `bins[{}]` must be an integer".format(i)) from e
```
inspired from the messages below.